tinygrad/codegen/uops.py

```diff
@@ -167,6 +167,7 @@ def remove_childless(self, keep:Set[UOp]):
         for vu in ru.vin:
           has_child.add(vu)
       nu: List[UOp] = [x for x in self.uops if x in has_child or x in keep]
+      self.saved_exprs = {k:v for k,v in self.saved_exprs.items() if v in nu}
```

Comment:
Does this have to be recomputed every time in the while loop, or is once at the end enough?